components/odh-notebook-controller/controllers/notebook_runtime.go

```diff
@@ -0,0 +1,238 @@
+				}
+			}
+		}
+	}
+
+	// Create or update the ConfigMap in the Notebook's namespace
+	configMap := &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      configMapName,
+			Namespace: notebookNamespace,
+			Labels:    map[string]string{"opendatahub.io/managed-by": "workbenches"},
+		},
+		Data: data,
+	}
+
+	// Check if the ConfigMap exists
+	existingConfigMap, configMapExists, err := r.getRuntimeConfigMap(ctx, configMapName, notebookNamespace)
+	if err != nil {
+		log.Error(err, "Error getting ConfigMap", "ConfigMap.Name", configMapName)
+		return err
+	}
+
+	if configMapExists {
+		if !jsonEqual(existingConfigMap.Data, data) {
+			existingConfigMap.Data = data
+			if err := r.Update(ctx, existingConfigMap); err != nil {
+				log.Error(err, "Failed to update ConfigMap", "ConfigMap.Name", configMapName)
+				return err
+			}
+			log.Info("Updated existing ConfigMap with new runtime images", "ConfigMap.Name", configMapName)
+		}
+	} else {
+		if err := r.Create(ctx, configMap); err != nil {
+			log.Error(err, "Failed to create ConfigMap", "ConfigMap.Name", configMapName)
+			return err
+		}
+		log.Info("Created new ConfigMap for runtime images", "ConfigMap.Name", configMapName)
+	}
+
+	return nil
+}
+
+// jsonEqual compares two JSON-like maps for equality.
+func jsonEqual(a, b map[string]string) bool {
+	aBytes, _ := json.Marshal(a)
+	bBytes, _ := json.Marshal(b)
+	return string(aBytes) == string(bBytes)
+}
+
+func extractDisplayName(metadata string) string {
+	var metadataMap map[string]interface{}
+	err := json.Unmarshal([]byte(metadata), &metadataMap)
+	if err != nil {
+		return ""
+	}
+	displayName, ok := metadataMap["display_name"].(string)
+	if !ok {
+		return ""
+	}
+	return displayName
+}
+
+func formatKeyName(displayName string) string {
+	replacer := strings.NewReplacer(" ", "-", "(", "", ")", "")
+	return strings.ToLower(replacer.Replace(displayName)) + ".json"
+}
+
+// parseRuntimeImageMetadata extracts the first object from the JSON array
+func parseRuntimeImageMetadata(rawJSON string) string {
+	var metadataArray []map[string]interface{}
+
+	err := json.Unmarshal([]byte(rawJSON), &metadataArray)
+	if err != nil || len(metadataArray) == 0 {
+		return "{}" // Return empty JSON object if parsing fails
+	}
+
+	// Convert first object back to JSON
+	metadataJSON, err := json.Marshal(metadataArray[0])
+	if err != nil {
+		return "{}"
+	}
+
+	return string(metadataJSON)
+}
+
+func (r *OpenshiftNotebookReconciler) EnsureNotebookConfigMap(notebook *nbv1.Notebook, ctx context.Context) error {
+	return r.syncRuntimeImagesConfigMap(ctx, notebook.Namespace, r.Namespace, r.Config)
+}
+
+func MountPipelineRuntimeImages(notebook *nbv1.Notebook, log logr.Logger) error {
+
+	// Define the volume
+	configVolume := corev1.Volume{
+		Name: volumeName,
+		VolumeSource: corev1.VolumeSource{
+			ConfigMap: &corev1.ConfigMapVolumeSource{
+				LocalObjectReference: corev1.LocalObjectReference{
+					Name: configMapName,
+				},
```

Comment:
```suggestion
				},
				Optional:             ptr.To(true),
```

we should include this volume as optional, 
- As this function runs on the webhook side, this gets injected even before the configmaps gets created
- as in future, let's assume users want to disable the OOTB runtimes images, then this come crucial